CMakeLists.txt

```diff
@@ -651,14 +653,59 @@ list(APPEND BUN_RAW_SOURCES "${BUN_WORKDIR}/codegen/WebCoreJSBuiltins.cpp")
 # endif()
 
 # --- Runtime.js ---
-if(NOT NO_CODEGEN)
-    add_custom_command(
-        OUTPUT "src/fallback.out.js"
-        WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
-        COMMAND "${ESBUILD}" "--target=esnext" "--bundle" "src/fallback.ts" "--format=iife" "--platform=browser" "--minify" "--outfile=src/fallback.out.js"
-        DEPENDS "src/fallback.ts"
-    )
-endif()
+#if(NOT NO_CODEGEN)
+#    add_custom_command(
+#        OUTPUT "src/fallback.out.js"
+#        WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+#        COMMAND "${ESBUILD}" "--target=esnext" "--bundle" "src/fallback.ts" "--format=iife" "--platform=browser" "--minify" "--outfile=src/fallback.out.js"
+#        DEPENDS "src/fallback.ts"
+#    )
+#endif()
+
+# --- Runtime JS ---
+add_custom_command(
+    OUTPUT "${BUN_SRC}/runtime.out.js"
+    COMMAND "${ESBUILD}" "--define:process.env.NODE_ENV=\"${NODE_ENV}\"" "--target=esnext" "--bundle" "--format=iife" "--platform=browser" "--global-name=BUN_RUNTIME" "--minify" "--external:/bun:*" "${BUN_SRC}/runtime/index.ts" "--outfile=${BUN_SRC}/runtime.out.js"
+    COMMAND cat ${BUN_SRC}/runtime.footer.js >> ${BUN_SRC}/runtime.out.js
+)
+add_custom_command(
+    OUTPUT "${BUN_SRC}/runtime.out.refresh.js"
+    COMMAND "${ESBUILD}" "--define:process.env.NODE_ENV=\"${NODE_ENV}\"" "--target=esnext" "--bundle" "--format=iife" "--platform=browser" "--global-name=BUN_RUNTIME" "--minify" "--external:/bun:*" "${BUN_SRC}/runtime/index-with-refresh.ts" "--outfile=${BUN_SRC}/runtime.out.refresh.js"
+    COMMAND cat ${BUN_SRC}/runtime.footer.with-refresh.js >> ${BUN_SRC}/runtime.out.refresh.js
+)
+add_custom_command(
+    OUTPUT "${BUN_SRC}/runtime.nodebun.pre.out.js"
```

Comment:
is this right?

Reply:
Yes this is an optimization. Current make file goes like this

build runtime.node.pre.out.js
cat runtime.node.pre.out.js runtime.footer.node.js > runtime.node.out.js

build runtime.bun.pre.out.js
cat runtime.bun.pre.out.js runtime.footer.bun.js > runtime.bun.out.js

build instructions for both runtime.node.pre.out.js and runtime.bun.pre.out.js are identical. So instead of building it twice we build it once and call the file runtime.nodebun.pre.out.js
and use it in the cats